model/common/src/icon4py/model/common/states/factory.py

```diff
@@ -500,7 +500,7 @@ def _compute(
     ) -> None:
         try:
             metadata = {v: factory.get(v, RetrievalType.METADATA) for k, v in self._output.items()}
-            dtype = metadata["dtype"]
+            dtype = metadata[next(iter(metadata))]["dtype"]
```

Comment:
is this really necessary, or is it because of some duplicated entries in the metadata (maybe the ones from the geometry_attributes? Can you test that?